Library/ViewModels/PledgeViewModel.swift

```diff
@@ -349,6 +293,44 @@ public class PledgeViewModel: PledgeViewModelType, PledgeViewModelInputs, Pledge
       applePayParams.wrapInOptional()
     )
 
+    // MARK: - Create Backing
+
+    let createBackingData = Signal.combineLatest(
+      project,
+      reward,
+      pledgeAmount,
+      selectedShippingRule,
+      selectedPaymentSourceId,
+      applePayParamsData,
+      refTag
+    )
+    .map { $0 as CreateBackingData }
+
+    let createButtonTapped = Signal.combineLatest(
+      self.submitButtonTappedSignal,
+      context
+    )
+    .filter { _, context in context.isCreating }
+    .ignoreValues()
+
+    let createBackingDataAndIsApplePay = createBackingData.takePairWhen(
+      Signal.merge(
+        createButtonTapped.mapConst(false),
+        willCreateApplePayBacking
+      )
+    )
+
+    let createBackingEvents = createBackingDataAndIsApplePay
+      .map(CreateBackingInput.input(from:isApplePay:))
+      .switchMap { input in
+        AppEnvironment.current.apiService.createBacking(input: input)
+          .ksr_delay(AppEnvironment.current.apiDelayInterval, on: AppEnvironment.current.scheduler)
+          .map { $0 as StripeSCARequiring }
```

Comment:
Moving this type mapping higher up means we don't have to do so many signal merges further down.